changelogs/unreleased/gh-9965-wrong-uint-comparison.md

```diff
@@ -0,0 +1,5 @@
+## bugfix/core
+
+- Fixed a bug in the unsigned numbers comparison in tuple keys. It allowed to
+  insert the same key multiple times into a unique index, and sometimes wouldn't
+  allow to find an existing key in an index (gh-9965).
```

Comment:
I'm not sure it fixes the issue for hash index. AFAIR we don't decode integers to compute the hash.

Reply:
Thanks for noticing! Indeed, hash index is also broken 🤯. We do decode, but only for indexes having a single key part with unsigned type. Otherwise fully broken.

I am currently investigating how to fix it with the smallest perf decrease.